lib/haml/railtie.rb

```diff
@@ -1,22 +1,19 @@
-if defined?(ActiveSupport)
-  require 'haml/template/options'
-  ActiveSupport.on_load(:before_initialize) do
-    ActiveSupport.on_load(:action_view) do
-      require "haml/template"
-    end
-  end
-end
+require 'haml/template/options'
 
 module Haml
   class Railtie < ::Rails::Railtie
     initializer :haml do |app|
-      require "haml/template"
-      if defined?(::Sass::Rails::SassTemplate) && app.config.assets.enabled
-        require "haml/sass_rails_filter"
+
+      ActiveSupport.on_load(:action_view) do
```

Comment:
Why is on_load hook inside initializer necessary? I guess either of on_load or initializer will be called first, then the other will be called. If they are always in the same order, either of them can be removed.